ios/Client/AppDelegate.swift

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+import Foundation
+
+@UIApplicationMain
+class AppDelegate: UMAppDelegateWrapper {
+  var rootViewController: EXRootViewController?
+
+  override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
+    if (application.applicationState != UIApplication.State.background) {
+      // App launched in foreground
+      setUpUserInterfaceForApplication(application, withLaunchOptions: launchOptions)
+    }
+
+    super.application(application, didFinishLaunchingWithOptions: launchOptions)
+
+    return true
+  }
+
+  override func applicationWillEnterForeground(_ application: UIApplication) {
+    setUpUserInterfaceForApplication(application, withLaunchOptions: nil)
+    super.applicationWillEnterForeground(application)
+  }
+
+  func setUpUserInterfaceForApplication(_ application: UIApplication, withLaunchOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]?) {
+    if (self.window != nil) {
+      return
+    }
```

Comment:
Can this method ever be called twice legitimately? Should we throw an error here if this happens?

Reply:
Yes, it can be called multiple times – it's called in `applicationWillEnterForeground` which of course can be called multiple times during the lifetime of the app.